packages/pipeline-editor/src/FileSubmissionDialog.tsx

```diff
@@ -28,186 +29,139 @@ interface IProps {
   schema: ISchema[];
 }
 
-interface IState {
-  displayedRuntimeOptions: IRuntime[];
-  includeDependency: boolean;
-  selectedRuntimePlatform: string;
-  validSchemas: ISchema[];
-}
-
-const EnvForm: React.FC<{ env: string[] }> = ({ env }) => {
-  if (env.length > 0) {
-    return (
-      <>
-        <br />
-        <br />
-        <div>Environment Variables:</div>
-        <br />
-        {Utils.chunkArray(env, 4).map((col, i) => (
-          <div key={i}>
-            {col.map(envVar => (
-              <div key={envVar}>
-                <label htmlFor={envVar}>{envVar}:</label>
-                <br />
-                <input
-                  type="text"
-                  id={envVar}
-                  className="envVar"
-                  name={envVar}
-                  size={30}
-                />
-              </div>
-            ))}
-          </div>
-        ))}
-      </>
-    );
-  }
-  return null;
-};
-
-export class FileSubmissionDialog extends React.Component<IProps, IState> {
-  state = {
-    displayedRuntimeOptions: new Array<IRuntime>(),
-    includeDependency: true,
-    selectedRuntimePlatform: '',
-    validSchemas: new Array<ISchema>()
-  };
+export const FileSubmissionDialog: React.FC<IProps> = ({
+  env,
+  dependencyFileExtension,
+  images,
+  runtimes,
+  schema
+}) => {
+  const validSchemas = React.useMemo(
+    (): ISchema[] => PipelineService.filterValidSchema(runtimes, schema),
+    [runtimes, schema]
+  );
 
-  handleCheck = (event: React.ChangeEvent<HTMLInputElement>): void => {
-    this.setState({
-      includeDependency: !this.state.includeDependency
-    });
-  };
+  const [includeDependency, setIncludeDependency] = React.useState(true);
+  const [platformSelection, setPlatformSelection] = React.useState(
+    (validSchemas && validSchemas[0]?.name) ?? ''
+  );
```

Comment:
this state is initially derived from a prop. If the prop changes, it won't be reflected in the state. I would keep the initial value separate and only combine them when needed:

```jsx
const defaultPlatform = validSchemas?.[0]?.name ?? ''
const [platformSelection, setPlatformSelection] = React.useState();

const resolvedPlatformSelection = platformSelection ?? defaultPlatform;
```